### PR TITLE
Add SDL2 port from Joel16

### PR DESCRIPTION
**Pull request by sharkwouter**
I'll leave this as a draft for now. This should probably include a patch file. I'm still not quite sure what the m4 files do, so I didn't include them.

### CODE REVIEW

**Comment by sharkwouter** [comment]
This is much more incomplete than I though. The build script will need to be changed to make cmake able to detect SDL2.

**Comment by sharkwouter** [comment]
This PR is probably good enough for use right now. It's not ideal because it doesn't package the files which can help build systems like cmake to find SDL2, but my attempts at using other build systems like autotools and cmake have been unsuccessful.

File: scripts/SDL2.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+download_and_extract https://github.com/joel16/SDL2/archive/2.0.9.tar.gz SDL2-2.0.9
+make -f Makefile.psp
+mkdir -p $(psp-config --psp-prefix)/include/SDL2
+cp -v include/*.h $(psp-config --psp-prefix)/include/SDL2
+cp -v libSDL2.a $(psp-config --psp-prefix)/lib
```
Comment (dbeef) [inline scripts/SDL2.sh:5]: Don't we need to copy `libSDL2main.a` as we are linking statically?

**Comment by dbeef** [comment]
As you are building with a custom PSP Makefile it's understandable that there is no cmake config script exported, the very same situation is with SDL 1.25, as it utilizes autogen to build artifacts.

Workaround for this situation in cmake is to just pass linking flags manually, as imported SDL target is unavailable:

```cmake
    target_link_libraries(Dependencies INTERFACE
            -L${PSPDEV}/psp/lib
            -lSDLmain
            -lSDL
            -lg
     ...
```

For me it's a good tradeoff to have SDL2, writing custom CMakeLists.txt just to build SDL2 is not worth it when we have this makefile already. You have my approve, good job, just address my comment above.

File: depends/check-SDL2.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+ PREFIX=$(psp-config --psp-prefix)
+
+ ls $PREFIX/lib/libSDL2.a $PREFIX/include/SDL2
```
Comment (dbeef) [inline depends/check-SDL2.sh:5]: Same question as below regarding `libSDL2main.a`

**Comment by dbeef** [comment]
After some off-github discussions I understand that the `libSDL2main.a` artifact is not created in this build process. 
This renders no problem, as defining `SDL_main` in one's own sources will provide an entry point for SDL2 anyway.

In such case, I am closing my discussions and merging.

**Comment by dbeef** [comment]
Hey @sharkwouter, I see the MR for `SDL2main` target has been merged:

https://github.com/joel16/SDL2/pull/1#event-3308529115

Could you please resolve the discussions above (copying `libSDL2main.a`)?

**Comment by sharkwouter** [comment]
Done, I've added SDL2main

**Comment by dbeef** [comment]
I rebuilt the toolchain from scratch for testing.
Used the following sources as a sample:

`CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.10)
add_executable(test-cmake main.cpp)

target_link_libraries(test-cmake PRIVATE SDL2main SDL2  ${PSP_LIBRARIES})

create_pbp_file(TARGET test-cmake)
```
`main.cpp`
```cpp
#include <SDL2/SDL.h>
extern "C"
{
	int SDL_main(int argc, char *argv[])
	{
		while(true){}
	}
}
```

Builds, links and runs on PPSSPP - merging.

**Pull request merged**